### PR TITLE
switch the request

### PR DESCRIPTION
**Pull request by pablopolyte**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Here we set a custom request for performance
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes {http://forge.prestashop.com/browse/EMKTG-210}.

### CODE REVIEW

File: src/Calculator/StatisticsCalculator.php [hunk 1]
```diff
@@ -227,6 +230,6 @@ public function computeConversionByProduct($id_product, $id_product_attribute, $
 
         $countAddedToWishlist = \Db::getInstance()->getValue($queryCountAll);
 
-        return round(($nbOrderPaidAndShipped / $countAddedToWishlist) * 100, 2);
+        return round(($nbOrderPaidAndShipped['nb'] / $countAddedToWishlist) * 100, 2);
```
Comment (PierreRambaud) [inline src/Calculator/StatisticsCalculator.php:233]: You should test if the key exists, I'm pretty sure it's not working when the database is empty ^^

Comment (pablopolyte) [reply src/Calculator/StatisticsCalculator.php:237]: Thank you

File: src/Calculator/StatisticsCalculator.php [hunk 2]
```diff
@@ -201,43 +201,25 @@ public function getProductImage($productDetails)
...
-        $queryCarts->where('id_product = ' . $id_product);
-        $queryCarts->where('id_product_attribute = ' . $id_product_attribute);
+        $nbOrderPaidAndShipped = [];
+        $queryOrders = '
+            SELECT count(distinct(o.id_order)) as nb
+            FROM ' . _DB_PREFIX_ . 'orders o
+            LEFT JOIN ' . _DB_PREFIX_ . 'blockwishlist_statistics bws ON (o.id_cart = bws.id_cart )
+            LEFT JOIN ' . _DB_PREFIX_ . 'order_history oh ON (o.`id_order` = oh.`id_order`)
+            LEFT JOIN ' . _DB_PREFIX_ . 'order_state os ON (os.`id_order_state` = oh.`id_order_state` AND os.`paid` = 1 AND os.`shipped` = 1)
+            LEFT JOIN ' . _DB_PREFIX_ . 'order_detail od ON (od.`id_order` = o.`id_order` AND od.`product_id` = ' . (int) $id_product . ' AND od.`product_attribute_id` = ' . (int) $id_product_attribute . ')
```
Comment (PierreRambaud) [inline src/Calculator/StatisticsCalculator.php:211]:
```suggestion
            LEFT JOIN ' . _DB_PREFIX_ . 'order_detail od ON (od.`id_order` = o.`id_order` AND od.`product_id` = bws.`id_product` AND od.`product_attribute_id` = bws.`id_product_attribute`'
```
You already test the product_attribute_id and product_id in the where clause.

File: src/Calculator/StatisticsCalculator.php [hunk 3]
```diff
@@ -201,43 +201,25 @@ public function getProductImage($productDetails)
...
-        $queryCarts->select('id_cart');
-        $queryCarts->from('blockwishlist_statistics');
-        $queryCarts->where('id_cart != 0');
-        $queryCarts->where('id_product = ' . $id_product);
-        $queryCarts->where('id_product_attribute = ' . $id_product_attribute);
+        $nbOrderPaidAndShipped = [];
+        $queryOrders = '
+            SELECT count(distinct(o.id_order)) as nb
+            FROM ' . _DB_PREFIX_ . 'orders o
+            LEFT JOIN ' . _DB_PREFIX_ . 'blockwishlist_statistics bws ON (o.id_cart = bws.id_cart )
```
Comment (PierreRambaud) [inline src/Calculator/StatisticsCalculator.php:208]:
```suggestion
            INNER JOIN ' . _DB_PREFIX_ . 'blockwishlist_statistics bws ON (o.id_cart = bws.id_cart )
```

**Review by PierreRambaud — APPROVED** [review]

**Comment by NeOMakinG** [comment]
Thanks @pablopolyte

**Pull request merged**